### ISSUE TITLE
RFC: Increase the connection timeout when making API request

### ISSUE THREAD

**Issue by manh-t** [nimblehq/android-templates#217]
## Issue

Our `CoroutineTemplate` has integrated Proxyman config which helps to update request/ response in API. But I'm facing the `SocketTimeoutException` when updating the request/ response via Proxyman.

## Solution

- Increase the timeout when making API requests so we have enough time to modify the payload.
- They are: `connectTimeout`, `readTimeout`, `writeTimeout` of `OkHttpClient`.
- Recommend increasing the timeout to `30 seconds`.

## Who Benefits?

Developers.

I think this is **essential**.

## What's Next?

- [ ] Increase the timeout for `connectTimeout`, `readTimeout` and `writeTimeout`.

**Comment by luongvo** [comment]
Voted 👍

**Comment by hoangnguyen92dn** [comment]
+1 👍

**Comment by Tuubz** [comment]
Approved 🚀 

Something extra to think about: Do we consider Proxyman as **essential** or **optional**?

**Comment by manh-t** [comment]
@Tuubz I think we should consider Proxyman as `essential` since we use it in most of our projects 🥼 You have integrated Proxyman in https://github.com/nimblehq/android-templates/pull/118 @Tuubz 😃

**Comment by Tuubz** [comment]
@manh-t Correct, I integrated Proxyman before we implemented the idea of essential/optional features 🏑 

A while back [not everyone in the team agreed that Proxyman should be essential](https://nimble-co.slack.com/archives/G41UCTGHL/p1642475278000300?thread_ts=1642413515.010600&cid=G41UCTGHL)

**Comment by Tuubz** [comment]
@thiennguyen0196 

> Increase the timeout when making API requests so we have enough time to modify the payload.

Just to clarify a bit here, whenever we use the "**breakpoint**" feature of Proxyman it cancels the breakpoint after 10 seconds. This is too little time for us to modify the JSON response, that's why we would like to increase it to 30 seconds 🙏 Would be nice if we could keep it at 10 seconds for release variants, but have it at 30 seconds for debug variants 💭 

Anyways, if you get stuck during the set-up of Proxyman. Please let me know, I'll help out 😁

**Comment by thiennguyen0196** [comment]
@Tuubz I will check it out and let you know if I stumble any problems 👌 I have got the idea of this issue and I will try to solve it. Many thanks for the presentation file you sent me, it helps a lot 😆

**Comment by luongvo** [comment]
@Tuubz is this still an RFC? I think we should remove RFC prefix after voting 🤓
